### PR TITLE
fix dark theme switching on initial welcome dialog

### PR DESCRIPTION
**Pull request by digant73**
**BUGFIXES:**
* Fixed broken switching to dark theme in case it is selected on the initial welcome dialog. Currently, dark theme is never engaged when selected on initial welcome dialog.
* Disabled `Use Dark Theme (Can Be Configured Later)` checkbox on welcome dialog opened from `Help->View The Welcome Dialog` menu (from that menu, the checkbox has no effect on switching theme; it can be changed only from `Config->GUI` tab)

### CODE REVIEW

File: rpcs3/rpcs3qt/main_window.cpp [hunk 1]
```diff
@@ -3275,6 +3275,8 @@ void main_window::CreateConnects()
 	connect(ui->welcomeAct, &QAction::triggered, this, [this]()
 	{
 		welcome_dialog* welcome = new welcome_dialog(m_gui_settings, true, this);
+
+		// welcome dialog is auto-flagged (in its constructor) as WA_DeleteOnClose, so its pointer is no more usable after welcome->open()
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:3279]: This is just plain wrong and it makes no sense even mentioning this here

File: rpcs3/rpcs3qt/gui_application.cpp [hunk 2]
```diff
@@ -163,16 +163,17 @@ bool gui_application::Init()
...
 		bool use_dark_theme = false;
 
-		connect(welcome, &QDialog::accepted, this, [&]()
+		connect(welcome, &QDialog::finished, this, [&]()
 		{
-			use_dark_theme = welcome->does_user_want_dark_theme();
+			use_dark_theme = welcome->use_dark_theme();
 		});
 
+		// welcome dialog is auto-flagged (in its constructor) as WA_DeleteOnClose, so its pointer is no more usable after welcome->exec()
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/gui_application.cpp:171]: While it's not necessarily wrong, I don't see the point in adding this comment

Comment (digant73) [reply rpcs3/rpcs3qt/gui_application.cpp:171]: Of course the code is clear to the main developers, but I usually always provide comments just to make clear for others (e.g. the need to use local variable `bool use_dark_theme` in `gui_application.cpp` that otherwise would be completely useless)

File: rpcs3/rpcs3qt/main_window.cpp (hunk 1 above)
Comment (digant73) [reply rpcs3/rpcs3qt/main_window.cpp:3279]: Yes, the comment should be more articulated for the `open()` method. It was provided more to highlight the fact that there was no need to handle the pointer deletion, but that is common to other dialogs opened in this source file. So simply removing it

File: rpcs3/rpcs3qt/gui_application.cpp (hunk 2 above)
Comment (Megamouse) [reply rpcs3/rpcs3qt/gui_application.cpp:171]: It actually doesn't really make sense to use the local variable in the first place.
Just move the guisettings call into the connect

File: rpcs3/rpcs3qt/gui_application.cpp [hunk 3]
```diff
@@ -161,19 +161,16 @@ bool gui_application::Init()
 	{
 		welcome_dialog* welcome = new welcome_dialog(m_gui_settings, false);
 
-		bool use_dark_theme = false;
-
-		connect(welcome, &QDialog::accepted, this, [&]()
+		connect(welcome, &QDialog::finished, this, [&]()
```
Comment (elad335) [inline rpcs3/rpcs3qt/gui_application.cpp:164]: Why the change to 'finished'? if the user rejects our terms then we should terminate execution instead.

Comment (digant73) [reply rpcs3/rpcs3qt/gui_application.cpp:164]: the signals in this `connect` and the `connect` in `welcome_dialog.cpp` needed to be swapped. Both are executed but the sequence is `accepted` (when you press on the `Continue` button) and then `terminated` (where the settings on the dialog are finally used). The bug was simply due to wrong signals registration.

Comment (elad335) [reply rpcs3/rpcs3qt/gui_application.cpp:164]: I see, I was specifically referring to the scenera in which the user closes the dialog forcefully, without agreeing to "I have read the Quickstart guide".
Can you add this code in this pr for the first time RPCS3 shows up?

connect(welcome, &QDialog::rejected, this, []() { std::exit(0); });

std::exit should be safe to use here.

Comment (digant73) [reply rpcs3/rpcs3qt/gui_application.cpp:164]: no pronlem to add that `connect`. But is that scenario related to the case where the rpcs3 app is forced to terminate when the welcome dialog is active? Current welcome dialog does not allow to close the dialog without accepting the agreement (the `continue` button is disabled)

Comment (Megamouse) [reply rpcs3/rpcs3qt/gui_application.cpp:164]: At this point you can just remove this connect all together and just move the guisettings call into the connect in the dialog

Comment (digant73) [reply rpcs3/rpcs3qt/gui_application.cpp:164]: yes, especially when other gui settings (e.g. `m_gui_settings->SetValue(gui::ib_show_welcome, QVariant(!checked));`) are managed directly on the dialog. Maybe, in the past, the logic was quite different needing a more articulated code.
Anyway, as last step from my side on this PR, just moving the logic on the dialog maintaining the compliancy to the current logic.
